traffic_monitor_golang/traffic_monitor/datareq/datareq.go

```diff
@@ -235,3 +276,39 @@ func addTrailingSlashEndpoints(dispatchMap map[string]http.HandlerFunc) map[stri
 	}
 	return dispatchMap
 }
+
+func acceptsGzip(r *http.Request) bool {
+	encodingHeaders := r.Header["Accept-Encoding"] // headers are case-insensitive, but Go promises to Canonical-Case requests
+	for _, encodingHeader := range encodingHeaders {
+		encodingHeader := strings.Replace(encodingHeader, " ", "", -1)
```

Comment:
This is incorrect. These headers have optional linear whitespace included, which can be spaces, tabs, or a CRLF, as long as it is followed by a space or a tab.